sonar-html-plugin/src/main/java/org/sonar/plugins/html/checks/sonar/LayoutTableWithSemanticMarkupCheck.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * SonarHTML :: SonarQube Plugin
+ * Copyright (c) 2010-2019 SonarSource SA and Matthijs Galesloot
+ * sonarqube@googlegroups.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sonar.plugins.html.checks.sonar;
+
+import static java.lang.String.format;
+
+import java.util.Optional;
+
+import org.sonar.check.Rule;
+import org.sonar.plugins.html.checks.AbstractPageCheck;
+import org.sonar.plugins.html.node.Attribute;
+import org.sonar.plugins.html.node.TagNode;
+
+@Rule(key = "S5258")
+public class LayoutTableWithSemanticMarkupCheck extends AbstractPageCheck {
+
+  private boolean isWithinLayoutTable = false;
+
+  @Override
+  public void startElement(TagNode node) {
+    if (isTable(node) && isLayout(node)) {
+      findAttribute(node, "SUMMARY").ifPresent(a -> createViolation(a.getLine(), format("Remove this \"%s\" attribute", a.getName())));
+      isWithinLayoutTable = true;
+    }
+    if (isWithinLayoutTable) {
+      if (isCaption(node) || isTableHeader(node)) {
+        createViolation(node.getStartLinePosition(), format("Remove this \"%s\" element", node.getNodeName()));
+      }
+      if (isTableColumn(node)) {
+        findAttribute(node, "HEADERS").ifPresent(a -> createViolation(a.getLine(), format("Remove this \"%s\" attribute", a.getName())));
+        findAttribute(node, "SCOPE").ifPresent(a -> createViolation(a.getLine(), format("Remove this \"%s\" attribute", a.getName())));
```

Comment:
Minor: this is more a matter of personal preference, but IMO I'd rather have lambdas slighly longer but with meaningful parameter names, such as `attribute` instead of `a`. But as this is a matter of preference, you can leave it as is if you prefer